### PR TITLE
[stdlib] Patch bool dtypepointer load

### PR DESCRIPTION
**Pull request by bgreni**
This PR offers a temporary fix to address #2813 in case the team cannot currently prioritize this, and I imagine a proper fix requires changes at a deeper level than OS contributors have access to?

### CODE REVIEW

**Comment by helehex** [comment]
Thanks for patching this

File: stdlib/test/memory/test_memory.mojo [hunk 1]
```diff
@@ -518,6 +518,25 @@ def test_indexing():
...
+    ptr.store(1, True)
+    ptr.store(2, True)
+    ptr.store(3, True)
+    assert_equal(
+        ptr.load[width=4](0), SIMD[DType.bool, 4](True, True, True, True)
+    )
+
+    var ptr2 = DTypePointer[DType.int32].alloc(4)
+    ptr2.store(0, 1)
+    ptr2.store(1, 2)
```
Comment (helehex) [inline stdlib/test/memory/test_memory.mojo:533]: Could you also add the inverse of this test, where you `store[width=4]()` and `load[width=1]()`?

**Comment by helehex** [comment]
i suppose there a chance the original behavior of packing the bits is desired? but im not sure.. we need some more input

**Comment by bgreni** [comment]
> I suppose there's a chance that the original behavior of packing the bits is desired?

I think that fact that my change works could point to `.bitcast[SIMD[DType.bool, width]]().load()` not working as intended? Regardless I think this code should work as expected and any special logic required to make it work should be implemented internally. Seems like a pretty major footgun that has no need to exist in a brand new language.

File: stdlib/test/memory/test_memory.mojo (hunk 1 above)
Comment (bgreni) [reply stdlib/test/memory/test_memory.mojo:533]: Good catch, `store[width=4]()` also doesn't work currently.

File: stdlib/src/memory/unsafe.mojo [hunk 2]
```diff
@@ -904,11 +904,25 @@ struct DTypePointer[
...
+        if type.is_bool():
+            var v = SIMD[type, width]()
+
+            @parameter
+            fn do_load[i: Int]():
+                v[i] = self.address.offset(int(offset) + i).load[
+                    alignment=alignment
+                ]()
+
+            unroll[do_load, width]()
```
Comment (soraros) [inline stdlib/src/memory/unsafe.mojo:918]:
```suggestion
            @parameter
            for i in range(width):
                v[i] = self.address.offset(int(offset) + i).load[
                    alignment=alignment
                ]()
```

Comment (bgreni) [reply stdlib/src/memory/unsafe.mojo]: Thanks for pointing that out!

**Comment by JoeLoser** [comment]
FYI @Dan13llljws in case you notice any issues as I know you're poking in this area

**Pull request closed without merging**